tools/cmake/FindBpfProg.cmake

```diff
@@ -0,0 +1,182 @@
+# SPDX-License-Identifier: GPL-2.0 OR BSD-3-Clause
+
+#[=======================================================================[.rst:
+FindBpfProg
+--------
+
+Find BpfProg
+
+This module finds if all the dependencies for eBPF Compile-Once-Run-Everywhere
+programs are available and where all the components are located.
+
+The caller may set the following variables to disable automatic
+search/processing for the associated component:
+
+  ``BPFTOOL_EXE``
+    Path to ``bpftool`` binary
+
+  ``CLANG_EXE``
+    Path to ``clang`` binary
+
+  ``LIBBPF_INCLUDE_DIRS``
+    Path to ``libbpf`` development headers
+
+  ``LIBBPF_LIBRARIES``
+    Path to `libbpf` library
+
+  ``GENERATED_VMLINUX``
+    Path to ``bpftool`` generated ``vmlinux.h`` file. If unset, this module will
```

Comment:
nit: either "bpftool-generated vmlinux.h" or "vmlinux.h generated by bpftool", currently it reads a bit awkward, IMO